File: archive/ruby/muse/algorithms/search/LinearSearchTest.rb
require "muse/algorithms/search/LinearSearch"
require "muse/util/SequenceBuilder"
require "muse/util/TestRunner"

def testLinearSearch
  size = 32768

  arr = Array.new(size)
  SequenceBuilder.packIncreasing(arr)

  return false if LinearSearch.find(arr, -1) != size

  return false if LinearSearch.find(arr, 2_147_483_647) != size

  arr.each_index do |i|
    return false if LinearSearch.find(arr, arr[i]) != i
  end

  true
end

TestRunner.pick(testLinearSearch) if __FILE__ == $PROGRAM_NAME

File: archive/nodejs/muse/algorithms/search/LinearSearchTest.js
import * as SequenceBuilder from '../../util/SequenceBuilder.js';
import * as TestRunner from '../../util/TestRunner.js';
import * as LinearSearch from './LinearSearch.js';

const testLinearSearch = () => {
  const size = 32768;

  const arr = new Array(size);
  SequenceBuilder.packIncreasing(arr);

  if (LinearSearch.find(arr, -1) !== size) {
    return false;
  }

  if (LinearSearch.find(arr, 2_147_483_647) !== size) {
    return false;
  }

  for (let i = 0; i < size; i++) {
    if (LinearSearch.find(arr, arr[i]) !== i) {
      return false;
    }
  }

  return true;
};

const main = () => {
  TestRunner.pick(testLinearSearch);
};

main();
